applications/StructuralMechanicsApplication/custom_elements/shell_thick_element_3D3N.cpp

```diff
@@ -0,0 +1,2801 @@
+			// http://kis.tu.kielce.pl/mo/COLORADO_FEM/colorado/IFEM.Ch31.pdf
+			// 
+			// Density and thickness are averaged over element.
+
+			// Average thickness over the whole element
+			double thickness = 0.0;
+			for (size_t i = 0; i < OPT_NUM_GP; i++)
+				thickness += mSections[i]->GetThickness();
+			thickness /= double(OPT_NUM_GP);
+
+			// Populate mass matrix with integation results
+			for (size_t row = 0; row < 18; row++)
+			{
+				if (row % 6 < 3)
+				{
+					// translational entry
+					for (size_t col = 0; col < 3; col++)
+					{
+						rMassMatrix(row, 6 * col + row % 6) = 1.0;
+					}
+				}
+				else
+				{
+					// rotational entry
+					for (size_t col = 0; col < 3; col++)
+					{
+						rMassMatrix(row, 6 * col + row % 6) = 
+							thickness*thickness / 12.0;
+					}
+				}
+
+				// Diagonal entry
+				rMassMatrix(row, row) *= 2.0;
+			}
+
+			rMassMatrix *= 
+				av_mass_per_unit_area*referenceCoordinateSystem.Area() / 12.0;
+		}// Consistent mass matrix
+		else
+		{
+			// Lumped mass matrix
+
+ 			// lumped area
+			double lump_area = referenceCoordinateSystem.Area() / 3.0;
+
+			// loop on nodes
+			for (size_t i = 0; i < 3; i++)
+			{
+				size_t index = i * 6;
+
+				double nodal_mass = av_mass_per_unit_area * lump_area;
+
+				// translational mass
+				rMassMatrix(index, index) = nodal_mass;
+				rMassMatrix(index + 1, index + 1) = nodal_mass;
+				rMassMatrix(index + 2, index + 2) = nodal_mass;
+
+				// rotational mass - neglected for the moment...
+			}
+		}// Lumped mass matrix
+	}
+
+	void ShellThickElement3D3N::CalculateDampingMatrix(MatrixType& rDampingMatrix, ProcessInfo& rCurrentProcessInfo)
+	{
+		if ((rDampingMatrix.size1() != OPT_NUM_DOFS) || (rDampingMatrix.size2() != OPT_NUM_DOFS))
+			rDampingMatrix.resize(OPT_NUM_DOFS, OPT_NUM_DOFS, false);
+
+		noalias(rDampingMatrix) = ZeroMatrix(OPT_NUM_DOFS, OPT_NUM_DOFS);
+	}
+
+	void ShellThickElement3D3N::CalculateLocalSystem(MatrixType& rLeftHandSideMatrix,
+		VectorType& rRightHandSideVector,
+		ProcessInfo& rCurrentProcessInfo)
+	{
+		CalculateAll(rLeftHandSideMatrix, rRightHandSideVector, rCurrentProcessInfo, true, true);
+	}
+
+	void ShellThickElement3D3N::CalculateRightHandSide(VectorType& rRightHandSideVector,
+		ProcessInfo& rCurrentProcessInfo)
+	{
+		Matrix dummy;
+		CalculateAll(dummy, rRightHandSideVector, rCurrentProcessInfo, true, true);
+	}
+
+	// =====================================================================================
+	//
+	// Class ShellThickElement3D3N - Results on Gauss Points
+	//
+	// =====================================================================================
+
+	void ShellThickElement3D3N::GetValueOnIntegrationPoints(const Variable<double>& rVariable,
+		std::vector<double>& rValues,
+		const ProcessInfo& rCurrentProcessInfo)
+	{
+		// resize output
+		if (rValues.size() != OPT_NUM_GP)
+			rValues.resize(OPT_NUM_GP);
+
+
+		int caseId = -1;
+		if (rVariable == TSAI_WU_RESERVE_FACTOR)
+		{
+			caseId = 10;
```

Comment:
It's for simplified control flow so large OR statements don't need to be repeated. After the initial OR statements which assign a caseID to groups of similar variables (i.e. VON_MISES_STRESS, VON_MISES_STRESS_TOP_SURFACE, VON_MISES_STRESS_MIDDLE_SURFACE and VON_MISES_STRESS_BOTTOM_SURFACE) they can all be distinguished downstream with just their caseID.

Nonetheless I cleaned up and condensed what was there anyway.